src/Engine/Data/Texture.cpp

```diff
@@ -97,7 +100,17 @@ void Texture::bindImageTexture( int unit,
 
 void Texture::updateData( void* newData ) {
     m_textureParameters.texels = newData;
-    needSync();
+
+    // register gpu task to update opengl representation before next rendering
+    static Core::TaskQueue::TaskId updateDataTaskId;
```

Comment:
should this be in critical section as several threads might call updateData concurrently ?

Reply:
I vote for re-entrant as the caller can be unaware of the existence of other threads nor how they modified data (e.g. Plugins)

Reply:
(btw static here is a bad idea, taskid should not be shared between instances of texture, I think this solve the problem)